scripts/nbconverted/latent_space_comparison.py

```diff
@@ -0,0 +1,320 @@
+vae_twolayer_file = '../tybalt/data/encoded_rnaseq_twohidden_100model.tsv.gz'
+vae_twolayer300_file = '../tybalt/data/encoded_rnaseq_twohidden_300model.tsv.gz'
+
+pca_encoded_df = pd.read_table(pca_file, index_col=0)
+ica_encoded_df = pd.read_table(ica_file, index_col=0)
+nmf_encoded_df = pd.read_table(nmf_file, index_col=0)
+adage_encoded_df = pd.read_table(adage_file, index_col=0)
+vae_encoded_df = pd.read_table(vae_file, index_col=0)
+vae_twolayer_encoded_df = pd.read_table(vae_twolayer_file, index_col=0)
+vae_twolayer300_encoded_df = pd.read_table(vae_twolayer300_file, index_col=0)
+
+
+# ## Process encoded feature data
+
+# In[9]:
+
+pca_ov_df, pca_ov_mean_df = get_encoded_ovsubtype_info(pca_encoded_df, ov_df)
+ica_ov_df, ica_ov_mean_df = get_encoded_ovsubtype_info(ica_encoded_df, ov_df)
+nmf_ov_df, nmf_ov_mean_df = get_encoded_ovsubtype_info(nmf_encoded_df, ov_df)
+adage_ov_df, adage_ov_mean_df = get_encoded_ovsubtype_info(adage_encoded_df, ov_df)
+vae_ov_df, vae_ov_mean_df = get_encoded_ovsubtype_info(vae_encoded_df, ov_df)
+vae_tl_ov_df, vae_tl_ov_mean_df = get_encoded_ovsubtype_info(vae_twolayer_encoded_df, ov_df)
+vae_tl300_ov_df, vae_tl300_ov_mean_df = get_encoded_ovsubtype_info(vae_twolayer300_encoded_df, ov_df)
+
+
+# ## HGSC Subtype Arithmetic
+# 
+# Because of the relationship observed in the consistent clustering solutions, perform the following subtraction:
+# 
+# _Immunoreactive_ - _Mesenchymal_
+# 
+# The goal is to observe the features with the largest difference and compare what the features represent depending on the dimensionality reduction algorithm
+# 
+# ### Part I. Visualizing feature activation differences across algorithms
+
+# In[10]:
+
+mes_immuno = ('Mesenchymal', 'Immunoreactive')
+
+
+# In[11]:
+
+algorithms = ['pca', 'ica', 'nmf', 'adage', 'tybalt', 'vae_100', 'vae_300']
+
+pca_ov_vector = ov_subtraction(pca_ov_mean_df, mes_immuno, 'pca')
+ica_ov_vector = ov_subtraction(ica_ov_mean_df, mes_immuno, 'ica')
+nmf_ov_vector = ov_subtraction(nmf_ov_mean_df, mes_immuno, 'nmf')
+adage_ov_vector = ov_subtraction(adage_ov_mean_df, mes_immuno, 'adage')
+vae_ov_vector = ov_subtraction(vae_ov_mean_df, mes_immuno, 'tybalt')
+vae_tl_ov_vector = ov_subtraction(vae_tl_ov_mean_df, mes_immuno, 'vae_100')
+vae_tl300_ov_vector = ov_subtraction(vae_tl300_ov_mean_df, mes_immuno, 'vae_300')
+
+
+# In[12]:
+
+latent_space_df = pd.concat([pca_ov_vector, ica_ov_vector,
+                             nmf_ov_vector, adage_ov_vector,
+                             vae_ov_vector, vae_tl_ov_vector,
+                             vae_tl300_ov_vector], axis=1)
+latent_space_df.head(2)
+
+
+# In[13]:
+
+long_latent_space_df = latent_space_df.stack().reset_index()
+long_latent_space_df.columns = ['rank', 'algorithm', 'activation']
+long_latent_space_df = long_latent_space_df[long_latent_space_df['algorithm'].isin(algorithms)]
+long_latent_space_df = long_latent_space_df.assign(algorithm_color =
```

Comment:
I would consider writing a version of this to file, but without the algorithm color and including the node/feature number.

Reply:
good point...I'd rather not have to rerun this script to fetch this data in the future...

Reply:
I think I want the color for consistency though :rainbow:

Reply:
Sure, but wouldn't you want to have the node number included in this file, too?

Reply:
:100:

Reply:
Node number added in 18221b1. Thanks!!